src/EditorFeatures/Core/Remote/SolutionChecksumUpdater.cs

```diff
@@ -202,57 +197,76 @@ await client.TryInvokeAsync<IRemoteAssetSynchronizationService>(
             cancellationToken).ConfigureAwait(false);
     }
 
-    private async ValueTask SynchronizeTextChangesAsync(
-        ImmutableSegmentedList<(Document oldDocument, Document newDocument)> values,
-        CancellationToken cancellationToken)
+    private void DispatchSynchronizeTextChanges(
+        Document oldDocument,
+        Document newDocument)
     {
-        var client = await RemoteHostClient.TryGetClientAsync(_workspace, cancellationToken).ConfigureAwait(false);
-        if (client == null)
-            return;
-
-        // this pushes text changes to the remote side if it can. this is purely perf optimization. whether this
-        // pushing text change worked or not doesn't affect feature's functionality.
-        //
-        // this basically see whether it can cheaply find out text changes between 2 snapshots, if it can, it will
-        // send out that text changes to remote side.
-        //
-        // the remote side, once got the text change, will again see whether it can use that text change information
-        // without any high cost and create new snapshot from it.
-        //
-        // otherwise, it will do the normal behavior of getting full text from VS side. this optimization saves
-        // times we need to do full text synchronization for typing scenario.
-        using var _ = ArrayBuilder<(DocumentId id, Checksum textChecksum, ImmutableArray<TextChange> changes, Checksum newTextChecksum)>.GetInstance(out var builder);
-
-        foreach (var (oldDocument, newDocument) in values)
+        // Attempt to inform the remote asset synchronization service as quickly as possible
+        // about the text changes between oldDocument and newDocument. By doing this, we can
+        // reduce the likelihood of the remote side encountering an unknown checksum and
+        // requiring a synchronization of the full document.
+        // This method uses JTF.RunAsync to create a fire-and-forget task. JTF.RunAsync will
+        // attempt to execute DispatchSynchronizeTextChangesHelperAsync synchronously if possible.
```

Comment:
But can this ever be synchronous?  It's an rpc call across processes after all.

Reply:
That's a good point, I'll update the comment. It can be sync right up to that RPC call (which is all we care about)